examples/python/Vision_files_preprocess.py

```diff
@@ -0,0 +1,623 @@
+    with open(header_name) as f:
+        data = f.read()
+
+    poss = re.search('matrix axis label\[2\]:=plane', data).span()
+    data = data.replace(data[poss[0]:poss[1]], \
+        'matrix axis label[2]:=sinogram views')
+
+    poss = re.search('matrix axis label\[3\]:=projection', data).span()
+    data = data.replace(data[poss[0]:poss[1]], \
+        'matrix axis label[3]:=number of sinograms')
+
+    with open(header_name_new, 'w') as f2:
+        f2.write(data)
+
+def change_max_ring_distance(header_name_new, header_name, max_ring_diff):
+    with open(header_name) as f:
+        data = f.read()
+
+    poss = re.search('%maximum ring difference\s*:=[^\n]*', data).span()
+    data = data.replace(data[poss[0]:poss[1]], \
+        '%maximum ring difference:={}'.format(int(max_ring_diff)))
+
+    with open(header_name_new, 'w') as f2:
+        f2.write(data)
+
+def remove_tof_dimension(header_name_new, header_name):
+    
+    with open(header_name) as f:
+        data = f.read()
+
+    poss = re.search(r'number of dimensions\s*:=[^\n]*', data).span()
+    data = data.replace(data[poss[0]:poss[1]], \
+        'number of dimensions:={}'.format(3))
+
+    data_type_string = 'matrix size\[4\]\s*:=[^\n]*\n'
+    data = re.sub(data_type_string, '', data)
+
+    data_type_string = 'matrix axis label\[4\]*\s*:=TOF bin*\n'
+    data = re.sub(data_type_string, '', data)
+
+    data_type_string = 'scale factor \(ps\/bin\) (.*)\n'
+    data = re.sub(data_type_string, '', data)
+
+    data_type_string = '%TOF mashing factor[^\n]*\s*:=\s*[^\n]*\n'
+    data = re.sub(data_type_string, '%TOF mashing factor :=0\n', data)
+
+    data_type_string = '%number of TOF time bins[^\n]*\s*:=\s*[^\n]*\n'
+    data = re.sub(data_type_string, '', data)
+
+    with open(header_name_new, 'w') as f2:
+        f2.write(data)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+                    prog='Vision_files_preprocess.py',
+                    description='Converts e7tools sinogram files for the Vision into sinogram files that can be read by STIR')
+    parser.add_argument('--prompts_filename_inclPath', required=True, help="The filename of the prompts file, including path")
+    parser.add_argument('--mu_map_header', default='umap_00.h33', help="The filename of the mu-map header")
+    parser.add_argument('--randoms_data_filename', default='smoothed_rand_00.s', help="The filename of the randoms data")
+    parser.add_argument('--scatter_2D_header_filename', default='scat_00_00.s.hdr', help="The filename of the 2D scatter header")
+    parser.add_argument('--norm_sino_data_filename', default='norm3d_00.a', help="The filename of the norm sinogram data")
+    parser.add_argument('--attenuation_corr_factor_data_filename', default='acf_00.a', help="The filename of the attenuation correction factor data")
+    parser.add_argument('--STIR_output_folder', default='processing', help="The folder where the output files are written to")
```

Comment:
This could be moved to the top (just before `def process_files`) if you want to avoid re-writing argument descriptions within `process_files` docstring.

Reply:
I think it's best to keep the docstrings in `process_files` as otherwise somebody who imports this won't know what to do. Not great for duplication, but so be it.

I don't have any opinion on keeping this at the bottom or top.

Reply:
ok. that'll be for later then, as this issue is also in some of the other python example files (they're generally in a bad state anyway). Thanks!